tools/ci/tc/decision.py

```diff
@@ -283,17 +283,27 @@ def add_task(task_name, task):
         task_id_map[task_name] = (task_id, task_data)
 
     for task_name, task in iteritems(tasks):
+        if task_name == "sink-task":
+            # sink-task will be created below at the end of the ordered dict,
+            # so that all other tasks will have been created.
+            continue
         add_task(task_name, task)
 
     # GitHub branch protection needs us to name explicit required tasks - which
     # doesn't suffice when using a dynamic task graph. To work around this we
     # declare a sink task that depends on all the other tasks completing, and
     # checks if they have succeeded. We can then make the sink task the sole
     # required task for GitHub.
-    depends_on_ids = [x[0] for x in task_id_map.values()]
-    sink_task = all_tasks['sink-task']
-    sink_task['command'] += ' {0}'.format(' '.join(depends_on_ids))
-    task_id_map['sink-task'] = create_tc_task(event, sink_task, taskgroup_id, depends_on_ids)
+    # sink-task is only scheduled on PRs.
```

Comment:
This could likely be incorporated into the above comment, i.e. something like:

```
    # GitHub branch protection for pull requests needs us to name explicit
    # required tasks - which doesn't suffice when using a dynamic task graph.
    # To work around this we declare a sink task that depends on all the other
    # tasks completing, and checks if they have succeeded. We can then
   # make the sink task the sole required task for pull requests.
```